src/ace_button/AceButton.h

```diff
@@ -279,7 +279,7 @@ class AceButton {
      * correctly, which may cause other event detection algorithms to fail.
      */
     void check();
-
+	void check(int input);
```

Comment:
I'm confused, why don't you call `checkState(uint8_t)` directly? It's already a public method.

By the way, your hack works, but it is not the recommended solution. The intention was that you would create a subclass of `ButtonConfig`, similar to `Encoded4To2ButonConfig`, and override the virtual `readButton()` method where the code would decode how your buttons are wired up to the PCF8574. Something like this:

```C++
class PcfButtonConfig : public ButtonConfig {
  public:
    int readButton(uint8_t pin) override {
      // extract the value of the button identified by the synthetic `pin` number
      // using the output from the PCF8574
      ...
      return {value of the button at the synthetic `pin` number};
    }
};
```

Then use `PcfButtonConfig` to configure one or more instances of `AceButton`, and call `AceButton::check()` on each button.

Reply:
**Addendum**: I'm not familiar with the `Adafruit_PCF8574.h` library, but from your sample code it looks like the decoding is already done in the library, so your `PcfButtonConfig::readButton()` method will be as simple as:

```C++
    int readButton(uint8_t pin) override {
      return pcf.digitalRead(pin));
    }
```

Reply:
Your'e right, I'm just rusty with C++. Trying to take a shortcut. Delete the pull request.